### PR TITLE
Fix 15798 add test for poly copy

### PR DESCRIPTION
**Pull request by galbwe**
This PR adds two tests to `test_polytools.py` that check that copied or deserialized polynomials are considered equal to the original. There are already some tests of a similar nature in [PR 15799](https://github.com/sympy/sympy/pull/15799), but suggested changes in that PR have gone unaddressed for over two weeks. 

#### References to other Issues or PRs

Addresses [Issue 15798](https://github.com/sympy/sympy/issues/15798)

#### Brief description of what is fixed or changed
Adds two tests to `test_polytools.py` named `test_poly_copy_equals_original` and `test_deserialized_poly_equals_original`. Both of these tests duplicate a `Poly` object, then check that it is equal to the original and has the same generators. 

#### Release Notes

NO ENTRY

### CODE REVIEW

File: sympy/polys/tests/test_polytools.py [hunk 1]
```diff
@@ -3358,3 +3360,21 @@ def test_issue_19360():
 
     f = -I*t*x - t*y + x*z - I*y*z
     assert factor(f, extension=I) == (x - I*y)*(-I*t + z)
+
+
+def test_poly_copy_equals_original():
+    poly = Poly(x + y, x, y, z)
+    copy = poly.copy()
+    assert poly == copy, (
+        "Copied polynomial not equal to original.")
```
Comment (czgdp1807) [inline sympy/polys/tests/test_polytools.py:3369]: Messages are good to add.

**Comment by czgdp1807** [comment]
ping @oscarbenjamin Can we merge this once tests pass?

**Comment by oscarbenjamin** [comment]
Yes this is fine to merge. The docs job has failed on Travis for an unrelated reason though.

**Comment by oscarbenjamin** [comment]
The test failure is due to #19551

**Comment by oscarbenjamin** [comment]
Closed and reopened to restart the tests now that #19551 is fixed.

**Pull request merged**